zerver/actions/message_edit.py

```diff
@@ -526,67 +633,21 @@ def do_update_message(
-            possible_stream_wildcard_mention=mention_data.message_has_stream_wildcards(),
-        )
+        if not target_message.is_stream_message():
+            update_edit_history(target_message, timestamp, edit_history_event)
 
-        event["online_push_user_ids"] = list(info.online_push_user_ids)
-        event["dm_mention_push_disabled_user_ids"] = list(info.dm_mention_push_disabled_user_ids)
-        event["dm_mention_email_disabled_user_ids"] = list(info.dm_mention_email_disabled_user_ids)
-        event["stream_push_user_ids"] = list(info.stream_push_user_ids)
-        event["stream_email_user_ids"] = list(info.stream_email_user_ids)
-        event["followed_topic_push_user_ids"] = list(info.followed_topic_push_user_ids)
-        event["followed_topic_email_user_ids"] = list(info.followed_topic_email_user_ids)
-        event["muted_sender_user_ids"] = list(info.muted_sender_user_ids)
-        event["prior_mention_user_ids"] = list(prior_mention_user_ids)
-        event["presence_idle_user_ids"] = filter_presence_idle_user_ids(info.active_user_ids)
-        event["all_bot_user_ids"] = list(info.all_bot_user_ids)
-        if rendering_result.mentions_stream_wildcard:
-            event["stream_wildcard_mention_user_ids"] = list(info.stream_wildcard_mention_user_ids)
-            event["stream_wildcard_mention_in_followed_topic_user_ids"] = list(
-                info.stream_wildcard_mention_in_followed_topic_user_ids
-            )
-        else:
-            event["stream_wildcard_mention_user_ids"] = []
-            event["stream_wildcard_mention_in_followed_topic_user_ids"] = []
+            # This does message.save(update_fields=[...])
+            save_message_for_edit_use_case(message=target_message)
 
-        if rendering_result.mentions_topic_wildcard:
-            event["topic_wildcard_mention_user_ids"] = list(info.topic_wildcard_mention_user_ids)
-            event["topic_wildcard_mention_in_followed_topic_user_ids"] = list(
-                info.topic_wildcard_mention_in_followed_topic_user_ids
-            )
-            topic_participant_user_ids = info.topic_participant_user_ids
-        else:
-            event["topic_wildcard_mention_user_ids"] = []
-            event["topic_wildcard_mention_in_followed_topic_user_ids"] = []
-            topic_participant_user_ids = set()
+            event["message_ids"] = update_message_cache([target_message])
+            users_to_be_notified = list(map(user_info, ums))
+            send_event_on_commit(user_profile.realm, event, users_to_be_notified)
 
-        update_user_message_flags(rendering_result, ums, topic_participant_user_ids)
-
-        do_update_mobile_push_notification(
-            target_message,
-            prior_mention_user_ids,
-            rendering_result.mentions_user_ids,
-            info.stream_push_user_ids,
-        )
+            changed_messages_count = 1
+            return UpdateMessageResult(
```

Comment:
Did this change, so that we can just return early for direct messages and we can easily handle different `message_edit_request` types for stream and direct messages.